src/pe/debug.rs

```diff
@@ -131,60 +326,127 @@ pub struct CodeviewPDB70DebugInfo<'a> {
 }
 
 impl<'a> CodeviewPDB70DebugInfo<'a> {
-    pub fn parse(bytes: &'a [u8], idd: &ImageDebugDirectory) -> error::Result<Option<Self>> {
+    pub fn parse(
+        bytes: &'a [u8],
+        idd: ImageDebugDirectoryIterator<'_>,
+    ) -> error::Result<Option<Self>> {
         Self::parse_with_opts(bytes, idd, &options::ParseOptions::default())
     }
 
     pub fn parse_with_opts(
         bytes: &'a [u8],
-        idd: &ImageDebugDirectory,
+        idd: ImageDebugDirectoryIterator<'_>,
         opts: &options::ParseOptions,
     ) -> error::Result<Option<Self>> {
-        if idd.data_type != IMAGE_DEBUG_TYPE_CODEVIEW {
-            // not a codeview debug directory
-            // that's not an error, but it's not a CodeviewPDB70DebugInfo either
-            return Ok(None);
-        }
+        let idd = idd.collect::<Result<Vec<_>, _>>()?;
+        let idd = idd
+            .iter()
+            .find(|idd| idd.data_type == IMAGE_DEBUG_TYPE_CODEVIEW);
 
-        // ImageDebugDirectory.pointer_to_raw_data stores a raw offset -- not a virtual offset -- which we can use directly
-        let mut offset: usize = match opts.resolve_rva {
-            true => idd.pointer_to_raw_data as usize,
-            false => idd.address_of_raw_data as usize,
-        };
+        if let Some(idd) = idd {
+            // ImageDebugDirectory.pointer_to_raw_data stores a raw offset -- not a virtual offset -- which we can use directly
+            let mut offset: usize = match opts.resolve_rva {
+                true => idd.pointer_to_raw_data as usize,
+                false => idd.address_of_raw_data as usize,
+            };
 
-        // calculate how long the eventual filename will be, which doubles as a check of the record size
-        let filename_length = idd.size_of_data as isize - 24;
-        if filename_length < 0 {
-            // the record is too short to be plausible
-            return Err(error::Error::Malformed(format!(
-                "ImageDebugDirectory size of data seems wrong: {:?}",
-                idd.size_of_data
-            )));
-        }
-        let filename_length = filename_length as usize;
+            // calculate how long the eventual filename will be, which doubles as a check of the record size
```

Comment:
it looks like whitespace error was introduced here? i'm surprised CI doesn't catch this via `rustfmt`?